lightwood/mixers/nn/nn.py

```diff
@@ -24,6 +26,7 @@ def __init__(self, dynamic_parameters, is_categorical_output=False):
         self.optimizer_class = None
         self.optimizer_args = None
         self.criterion = None
+        self.sampler = None
```

Comment:
do we really ant to serialize this on model save, i dont think I am understanding this PR @George3d6

Reply:
I guess it doesn't need to be saved with the model, I didn't realize that implication of making it part of `self`, I just wanted it to be used by both the `iter_fit` and `error` methods without having to re-create it each time.

I will explain the purpose of the PR bellow.